### PR TITLE
test: add Encode operand test case

### PR DESCRIPTION
**Pull request by junbeomlee**
resolved: #149 

details:

add Encode operand test case
- [x] Test case

### CODE REVIEW

File: encoding/encode_test.go [hunk 1]
```diff
@@ -16,9 +16,91 @@
...
+	for i, test := range tests {
+		op := test.operand
+		bytecode, err := encoding.EncodeOperand(op)
+
+		if !bytes.Equal(bytecode, test.expectedByte) {
+			t.Fatalf("test[%d] - EncodeOperand() result wrong. expectedByte=%x, got=%x", i, test.expectedByte, bytecode)
+		}
 
+		switch err.(type) {
+		case nil:
```
Comment (hihiboss) [inline encoding/encode_test.go]:
```go
if err != test.expectedErr {
     t.Fatalf("~~~")
}

if err != nil {
     if err.Error() != test.expectedErr.Error() {
          t.Fatalf("~~~")
     }
}
```

plz consider about this!

But, I think that yours or this seem to be the best.

**Comment by hihiboss** [comment]
```go
switch err.(type) {
	case nil:
		if err != test.expectedErr {
			t.Fatalf("test[%d] - EncodeOperand() result wrong. expectedByte=%x, got=%x", i, test.expectedByte, bytecode)
		}
	case error:
		if err.Error() != test.expectedErr.Error() {
			t.Fatalf("test[%d] - EncodeOperand() result wrong. expectedByte=%x, got=%x", i, err.Error(), test.expectedErr.Error())
		}
	}
}
```

The log should be changed to `"EncodeOperand() result wrong. expectedErr=%x, got=%x"`

File: encoding/encode_test.go (hunk 1 above)
Comment (zeroFruit) [reply encoding/encode_test.go]: +1

Comment (zeroFruit) [reply encoding/encode_test.go]: or
```
if err != nil && err != test.expectedErr {
  ...
  continue
}
```

File: encoding/encode_test.go [hunk 2]
```diff
@@ -16,9 +16,91 @@
...
+			if err != test.expectedErr {
+				t.Fatalf("test[%d] - EncodeOperand() result wrong. expectedByte=%x, got=%x", i, test.expectedByte, bytecode)
```
Comment (zeroFruit) [inline encoding/encode_test.go]: t.Fatalf(`test[%d] - EncodeOperand() result wrong. expectedByte="%x", got="%x"`, ...) can be more readable! 👍

**Comment by junbeomlee** [comment]
i have updated!

**Review by zeroFruit — APPROVED** [review]

**Pull request merged**